src/main/java/com/google/devtools/build/lib/remote/Retrier.java

```diff
@@ -136,6 +137,43 @@ public void recordFailure() {}
         public void recordSuccess() {}
       };
 
+  public static class FailureRateCircuitBreaker implements CircuitBreaker {
+
+    private static final int MIN_NUMBER_OF_EXECUTIONS = 5;
```

Comment:
Given that during a build we'll have tens of thousands of requests, this seems like a small number. In the worst-case scenario, if there is 1 failure out from 5 executions, then the failure rate will be 20% and the cache will start rejecting calls. I think setting this to 100 or some bigger number will strike a better balance between giving up too quickly vs trying too much and giving up too late.

Reply:
👍